src/api/store/media_library.py

```diff
@@ -0,0 +1,123 @@
+from typing import Tuple
+
+from django.core.exceptions import ValidationError
+from database.models import Community, Media, UserMediaUpload, UserProfile
+from django.db.models import Q
+import time
+
+limit = 30
+
+
+class MediaLibraryStore:
+    def __init__(self):
+        self.name = "MediaLibrary Store/DB"
+
+    def fetch_content(self, args):
+        com_id = args["community_id"]
+        upper_limit = lower_limit = images = None
+        try:
+            upper_limit = args["upper_limit"]
+            lower_limit = args["lower_limit"]
+        except KeyError:
+            pass
+        if not upper_limit and not lower_limit:
+            images = Media.objects.filter(
+                Q(events__community__id=com_id)
+                | Q(actions__community__id=com_id)
+                | Q(testimonials__community__id=com_id)
+                | Q(user_upload__community__id=com_id)
+            ).order_by("-id")[:limit]
+        else:
+            images = (
+                Media.objects.filter(
+                    Q(events__community__id=com_id)
+                    | Q(actions__community__id=com_id)
+                    | Q(testimonials__community__id=com_id)
+                    | Q(user_upload__community__id=com_id)
+                )
+                .exclude(
+                    id__gte=lower_limit, id__lte=upper_limit
+                )  # exclude content that have already been retrieved
+                .order_by("-id")[:limit]
+            )
+        return images, None
+
+    def generateQueryWithScope(self, scope, com_id):
+        queries = {
+            "actions": Media.objects.filter(actions__community__id=com_id),
+            "events": Media.objects.filter(events__community__id=com_id),
+            "testimonials": Media.objects.filter(testimonials__community__id=com_id),
+            "uploads": Media.objects.filter(user_upload__community__id=com_id),
+            "default": Media.objects.filter(
```

Comment:
why do the same job twice?  the queries are already expensive as they are.

You may want to do 
a = ...
b = ....
c = ....
d = ...

default = a|b|c|d

You get the idea?

Reply:
I read that when you write queries like this, django actually doesnt fetch anything, until you use it. When you write it like this, it actually just builds only the queries. 🤔

Reply:
Also I dont think `a|b...` fits in this particular scenario, cos we want to only be finding images related to "actions" when we specify the scope as "actions" 🤔